main.go

```diff
@@ -10,10 +10,23 @@ import (
 	"os/signal"
 	"strings"
 	"syscall"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
 var gitCommit = "n/a"
 
+var (
+	httpRequestsTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_requests_total",
+			Help: "Total number of HTTP requests",
+		},
+		[]string{"path"},
```

Comment:
I think the prometheus sdk will prevent you from using path, you should use code here. They added such checks for cardinality reasons